Machine Learning/Regression Loss Functions/regression-loss-functions.ipynb

```diff
@@ -0,0 +1,415 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "a = symbols('a') #actual value\n",
+    "p = symbols('p') #predicted value\n",
+    "mse = lambda a,p: (a-p)**2\n",
+    "mse_plot = plot(mse(0, p),(p, -3, 3), show=False, legend=True, line_color=\"red\")\n",
+    "mse_plot[0].label='MSE'\n",
+    "mse_plot.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The problem with the squared error is that it makes small errors (< 1.0) even smaller and large large errors (>1.0) disproportionately larger.\n",
+    "\n",
+    "This means that if you have a few outliers in your data, i.e. due to a large measurement error, the model will put useless effort to fit these data points and other points will fall by the wayside. \n",
+    "\n",
+    "So what you often want to do is instead of fitting the mean **squared** error to fit the mean **absolute** error (MAE)\n",
```

Comment:
I think I'd rephrase this to say that in this case, i.e. when you suspect outliers, since we're saying by default use MSE:
```suggestion
    "In these cases what you would usually want to do is instead of fitting the mean **squared** error to fit the mean **absolute** error (MAE)\n",
```

Reply:
It might also be worth saying you can check the model for outliers after fitting by plotting a histogram of prediction errors to see if it might be worth changing loss.

Reply:
I also added a section on residual distrbution